### DISCUSSION TITLE
How to include graphql-scalars?

### DISCUSSION THREAD

**Discussion by tsongas** [dotansimha/graphql-code-generator#7572] [Q&A] [answered]
I'm trying to add `graphql-scalars` to my project, but in my `codegen.yml` file I've got `schema: './src/schema/**/*.ts'` because I'm using `graphql-codegen --watch` to regenerate types whenever my a schema file changes, which I've found super convenient.

The problem is, `graphql-scalars` obviously doesn't live in my `.src/schema` folder, so GraphQL Code Generator doesn's see it. I've looked at the docs [here](https://www.graphql-code-generator.com/docs/config-reference/schema-field) and I see how to include multiple schema paths, but I don't see an example of how to include a module like `graphql-scalars`. Any suggestions appreciated!

**Comment by tsongas** [comment, marked as answer]
I got some help on Discord, apparently I needed to add references to the scalars, just importing graphql-scalars didn't do that, so I added a scalar.ts file similar to my enum.ts file for enums that gets imported into my schema, and now these scalars work:
```
import { gql } from 'apollo-server-express'

export const typeDef = gql`
  scalar Date
  scalar DateTime
  scalar Time
  scalar UUID
`
```

**Reply by Urigo** [reply]
@tsongas is there anything we can do on the docs or library to make it easier for others?

**Reply by tsongas** [reply]
@Urigo I'm sure showing this in the documentation would help, I don't think it was mentioned anywhere or I wouldn't have had the problem. Of course I'm new to GraphQL and Apollo Server so others may take for granted what isn't obvious to me.

**Reply by Urigo** [reply]
@tsongas where would you add it?
Maybe you can send us a PR with a suggestion on how and where would be best to do it from your perspective?

**Reply by ifeanyi-ugwu** [reply]
@Urigo, supporting @tsongas's answer, I have a tiny [utility](https://www.npmjs.com/package/graphql-scalars-type-defs-extractor) that I use in all the graphql servers that I build. Let me know what you think
